app/src/main/java/org/dhis2/utils/granularsync/SMSSyncProviderImpl.kt

```diff
@@ -0,0 +1,165 @@
+package org.dhis2.utils.granularsync
+
+import io.reactivex.Completable
+import io.reactivex.Single
+import io.reactivex.observers.DisposableCompletableObserver
+import io.reactivex.observers.DisposableSingleObserver
+import java.util.Date
+import org.dhis2.commons.resources.ResourceManager
+import org.dhis2.usescases.sms.SmsSendingService
+import org.hisp.dhis.android.core.D2
+import org.hisp.dhis.android.core.sms.domain.interactor.SmsSubmitCase
+import org.hisp.dhis.android.core.sms.domain.repository.SmsRepository
+import timber.log.Timber
+
+class SMSSyncProviderImpl(
```

Comment:
unit tests?

Reply:
There are unit test already in the SMSSyncProviderTest.kt class